### PR TITLE
nfs: remove secret, configmap when downscaling CephNFS

### PR DESCRIPTION
**Pull request by BlaineEXE**
When downscaling CephNFS clusters, be sure to remove secrets and
configmaps. Also be sure to only downscale for the currently-reconciling
CephNFS cluster.

Signed-off-by: Blaine Gardner <blaine.gardner@redhat.com>

**NOTE:** actually a fairly small PR except the massively expanded unit test

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: If this is only a documentation change, add the label `skip-ci` on the PR.
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/development-flow.html#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: pkg/operator/ceph/nfs/nfs.go [hunk 1]
```diff
@@ -219,27 +222,39 @@ func (r *ReconcileCephNFS) createConfigMap(n *cephv1.CephNFS, name string) (stri
 
 // Down scale the ganesha server
 func (r *ReconcileCephNFS) downCephNFS(n *cephv1.CephNFS, nfsServerListNum int) error {
-	diffCount := nfsServerListNum - n.Spec.Server.Active
-	for i := 0; i < diffCount; {
-		depIDToRemove := nfsServerListNum - 1
+	for i := nfsServerListNum - 1; i >= n.Spec.Server.Active; i-- {
+		idToRemove := i
```
Comment (BlaineEXE) [inline pkg/operator/ceph/nfs/nfs.go:227]: Reworked this loop to run backward from high to low IDs. There was no reason for the loop to be so complicated as to calculate `diffCount` and then count upwards from there. Or to track `i++` separately from the loop definition.

File: pkg/operator/ceph/nfs/controller.go [hunk 2]
```diff
@@ -312,28 +312,28 @@ func (r *ReconcileCephNFS) reconcileCreateCephNFS(cephNFS *cephv1.CephNFS) (reco
...
+	listOps := metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("%s=%s,%s=%s", k8sutil.AppAttr, AppName, CephNFSNameLabelKey, cephNFS.Name),
+	}
+	deployments, err := r.context.Clientset.AppsV1().Deployments(cephNFS.Namespace).List(r.opManagerContext, listOps)
+	if err != nil && !kerrors.IsNotFound(err) {
+		return reconcile.Result{}, errors.Wrapf(err, "failed to list deployments for CephNFS %q", cephNFS.Name)
+	}
+	currentNFSServerCount := 0
+	if deployments != nil {
+		currentNFSServerCount = len(deployments.Items)
```
Comment (BlaineEXE) [inline pkg/operator/ceph/nfs/controller.go:325]: Code used to list **all** NFS deployments, even those not part of the current NFS cluster, so fix that.

File: pkg/operator/ceph/nfs/controller_test.go [hunk 3]
```diff
@@ -116,113 +213,260 @@ func TestCephNFSController(t *testing.T) {
...
+			assert.NoError(t, err)
+			assert.False(t, res.Requeue)
+			assertCephNFSReady(t, r)
+			assertResourcesExist(t, cCtx, "rook-ceph-nfs-my-nfs-a")
+			assert.Len(t, *deploymentsUpdated, 1)
+			assert.Equal(t, "rook-ceph-nfs-my-nfs-a", (*deploymentsUpdated)[0].Name)
+		})
+	})
+
+	t.Run("multiple CephNFS clusters", func(t *testing.T) {
```
Comment (BlaineEXE) [inline pkg/operator/ceph/nfs/controller_test.go:407]: Make sure we can run multiple and that downscaling one doesn't downscale another.

**Comment by BlaineEXE** [comment]
setting `do-not-merge` while I finish testing

**Review by travisn — COMMENTED** [review]
LGTM! I'll approve after your done testing...

**Comment by BlaineEXE** [comment]
Done testing. I will leave removing keyring secrets for its own follow-up PR per issue https://github.com/rook/rook/issues/9860

**Review by travisn — APPROVED** [review]

**Pull request merged**